src/FloatingMenuPlugin.ts

```diff
@@ -0,0 +1,570 @@
+  if (plugin._popUpHandle) {
+    plugin._popUpHandle.update({
+      ...plugin._popUpHandle['props'],
+      pasteAsReferenceEnabled: true,
+    });
+  }
+  if (plugin._popUpHandle?.close) {
+    plugin._popUpHandle.close(null);
+    plugin._popUpHandle = null;
+  }
+}
+
+export function createSliceObject(editorView: EditorView): SliceModel {
+  const instanceUrl = 'http://modusoperandi.com/editor/instance/';
+  const referenceUrl = 'http://modusoperandi.com/ont/document#Reference_nodes';
+
+  const sliceModel: SliceModel = {
+    name: '',
+    description: '',
+    id: '',
+    referenceType: '',
+    source: '',
+    from: '',
+    to: '',
+    ids: [],
+  };
+
+  const objectIds: string[] = [];
+  let firstParagraphText: string | null = null;
+
+  editorView.focus();
+
+  const $from = editorView.state.selection.$from;
+  const $to = editorView.state.selection.$to;
+
+  const from = $from.start($from.depth);
+  const to = $to.end($to.depth);
+
+  editorView.state.doc.nodesBetween(from, to, (node) => {
+    if (node.type.name === 'paragraph') {
+      if (!firstParagraphText && node.textContent?.trim()) {
+        firstParagraphText = node.textContent.trim();
+      }
+      if (node.attrs?.objectId) {
+        objectIds.push(node.attrs.objectId);
+      }
+    }
+  });
+
+  sliceModel.id = instanceUrl + uuidv4();
+  sliceModel.ids = objectIds;
+  sliceModel.from = objectIds.length > 0 ? objectIds[0] : '';
+  sliceModel.to = objectIds.length > 0 ? objectIds[objectIds.length - 1] : '';
+
+  const viewWithDocView = editorView as EditorView;
+  sliceModel.source = viewWithDocView?.['docView']?.node?.attrs?.objectId;
+  sliceModel.referenceType = referenceUrl;
+
+  const today = new Date().toISOString().split('T')[0];
+  const snippet = (firstParagraphText || 'Untitled').substring(0, 20);
+  sliceModel.name = `${snippet} - ${today}`;
+
+  return sliceModel;
+}
+
+export function copySelectionPlain(
+  view: EditorView,
+  plugin: FloatingMenuPlugin
+) {
+  if (!view.hasFocus()) {
+    view.focus();
+  }
+  const { from, to } = view.state.selection;
+  if (from === to) return;
+
+  const slice = view.state.doc.slice(from, to);
+  const text = slice.content.textBetween(0, slice.content.size, '\n');
+
+  navigator.clipboard
+    .writeText(text)
+    .then(() => { }) //console.log('Plain text copied!'))
+    .catch((err) => console.error('Clipboard write failed:', err));
+  if (plugin._popUpHandle?.close) {
+    plugin._popUpHandle.close(null);
+    plugin._popUpHandle = null;
+  }
+}
+
+export async function pasteFromClipboard(
+  view: EditorView,
+  plugin: FloatingMenuPlugin
+) {
+  try {
+    if (!view.hasFocus()) view.focus();
+
+    const text = await navigator.clipboard.readText();
+    let tr: Transaction;
+
+    try {
+      // Try parsing as JSON slice
+      const parsed = JSON.parse(text);
+      const slice = Slice.fromJSON(view.state.schema, parsed);
+      tr = view.state.tr.replaceSelection(slice);
+    } catch (jsonErr) {
```

Comment:
Sonar: Handle this exception or don't catch it at all.
```suggestion
    } catch {
```